x-pack/plugins/upgrade_assistant/public/application/components/overview/logs_step/logs_step.tsx

```diff
@@ -0,0 +1,224 @@
+    {
+      defaultMessage: 'View logs',
+    }
+  ),
+  enableLogsButtonLabel: i18n.translate(
+    'xpack.upgradeAssistant.overview.logsStep.enableLogsButtonLabel',
+    {
+      defaultMessage: 'Enable logs',
+    }
+  ),
+  logsCountDescription: (deprecationCount: number, checkpoint: string) => (
+    <FormattedMessage
+      id="xpack.upgradeAssistant.overview.logsStep.countDescription"
+      defaultMessage="You have {deprecationCount, plural, =0 {no} other {{deprecationCount}}} deprecation {deprecationCount, plural, one {issue} other {issues}} since {checkpoint}."
+      values={{
+        deprecationCount,
+        checkpoint: (
+          <>
+            <FormattedDate value={checkpoint} year="numeric" month="long" day="2-digit" />{' '}
+            <FormattedTime value={checkpoint} timeZoneName="short" hour12={false} />
+          </>
+        ),
+      }}
+    />
+  ),
+  missingPrivilegesTitle: i18n.translate(
+    'xpack.upgradeAssistant.overview.logsStep.missingPrivilegesTitle',
+    {
+      defaultMessage: 'You require index privileges to analyze the deprecation logs',
+    }
+  ),
+  loadingError: i18n.translate('xpack.upgradeAssistant.overview.logsStep.loadingError', {
+    defaultMessage: 'An error occurred while retrieving the count of deprecation logs',
+  }),
+  retryButton: i18n.translate('xpack.upgradeAssistant.overview.logsStep.retryButton', {
+    defaultMessage: 'Try again',
+  }),
+};
+
+interface LogStepProps {
+  setIsComplete: (isComplete: boolean) => void;
+  hasPrivileges: boolean;
+  navigateToEsDeprecationLogs: () => void;
+}
+
+const LogStepDescription = () => (
+  <EuiText>
+    <p>{i18nTexts.logsStepDescription}</p>
+  </EuiText>
+);
+
+const LogsStep = ({ setIsComplete, hasPrivileges, navigateToEsDeprecationLogs }: LogStepProps) => {
+  const {
+    services: { api },
+  } = useAppContext();
+
+  const { isDeprecationLogIndexingEnabled } = useDeprecationLogging();
+
+  const checkpoint = loadLogsCheckpoint();
+
+  const {
+    data: logsCount,
+    error,
+    isLoading,
+    resendRequest,
+    isInitialRequest,
+  } = api.getDeprecationLogsCount(checkpoint);
+
+  useEffect(() => {
+    if (!isDeprecationLogIndexingEnabled) {
+      setIsComplete(false);
+    }
+
+    setIsComplete(logsCount?.count === 0);
+
+    // Depending upon setIsComplete would create an infinite loop.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isDeprecationLogIndexingEnabled, logsCount]);
+
+  if (hasPrivileges === false && isDeprecationLogIndexingEnabled) {
+    return (
+      <>
+        <LogStepDescription />
+
+        <EuiSpacer />
+
+        <EuiCallOut
+          iconType="help"
+          color="warning"
+          title={i18nTexts.missingPrivilegesTitle}
+          data-test-subj="missingPrivilegesCallout"
+        />
+
+        <EuiSpacer />
+
+        <EuiButton onClick={navigateToEsDeprecationLogs} data-test-subj="viewLogsLink">
```

Comment:
if a user doesn't have index privileges, the only thing they can do in the `es deprecation logs page` is enable/disable log collection right? Wondering if it's worthy to still show this CTA in that case 🤔

Reply:
That's correct. I was on the fence for this as well. I'm going to go ahead and remove it.

I looked back at the previous code when it was part of the deprecations step, and I see we weren't showing the sentence/link to the deprecation logs when a user did not have privileges.

Reply:
<img width="1218" alt="Screen Shot 2022-03-01 at 9 36 26 AM" src="https://user-images.githubusercontent.com/5226211/156188699-35929729-7f1f-44c3-8088-cf36b3c1f772.png">